app/shared/cms/haldoc.ts

```diff
@@ -1,98 +1,165 @@
-import {Http, Headers} from 'angular2/http';
 import {Observable} from 'rxjs/Observable';
-import TemplateParser from 'url-template';
+import {HalRemote} from './halremote';
+
+// Bring observables up to snuff
+export interface HalObservable<T> extends Observable<T> {
+  follow(rel: string, params?: {}): HalObservable<T>;
+  followList(rel: string, params?: {}): HalObservable<T[]>;
+  followItems(rel: string, params?: {}): HalObservable<T[]>;
+}
+Observable.prototype['follow'] = function(rel: string, params: {} = null) {
```

Comment:
You're adding these follow* methods to Observable itself? Is this b/c you can't subtype it or something?